native/spark-expr/src/math_funcs/abs.rs

```diff
@@ -0,0 +1,879 @@
+                        } else {
+                            Err(arithmetic_overflow_error("Int64").into())
+                        }
+                    }
+                },
+            },
+            ScalarValue::Float32(a) => Ok(ColumnarValue::Scalar(ScalarValue::Float32(
+                a.map(|x| x.abs()),
+            ))),
+            ScalarValue::Float64(a) => Ok(ColumnarValue::Scalar(ScalarValue::Float64(
+                a.map(|x| x.abs()),
+            ))),
+            ScalarValue::Decimal128(a, precision, scale) => match a {
+                None => Ok(args[0].clone()),
+                Some(v) => match v.checked_abs() {
+                    Some(abs_val) => Ok(ColumnarValue::Scalar(ScalarValue::Decimal128(
+                        Some(abs_val),
+                        *precision,
+                        *scale,
+                    ))),
+                    None => {
+                        if !fail_on_error {
+                            // return the original value
+                            Ok(ColumnarValue::Scalar(ScalarValue::Decimal128(
+                                Some(*v),
+                                *precision,
+                                *scale,
+                            )))
+                        } else {
+                            Err(arithmetic_overflow_error("Decimal128").into())
+                        }
+                    }
+                },
+            },
+            ScalarValue::Decimal256(a, precision, scale) => match a {
+                None => Ok(args[0].clone()),
+                Some(v) => match v.checked_abs() {
+                    Some(abs_val) => Ok(ColumnarValue::Scalar(ScalarValue::Decimal256(
+                        Some(abs_val),
+                        *precision,
+                        *scale,
+                    ))),
+                    None => {
+                        if !fail_on_error {
+                            // return the original value
+                            Ok(ColumnarValue::Scalar(ScalarValue::Decimal256(
+                                Some(*v),
+                                *precision,
+                                *scale,
+                            )))
+                        } else {
+                            Err(arithmetic_overflow_error("Decimal256").into())
+                        }
+                    }
+                },
+            },
+            dt => exec_err!("Not supported datatype for ABS: {dt}"),
+        },
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use datafusion::common::cast::{
+        as_decimal128_array, as_decimal256_array, as_float32_array, as_float64_array,
+        as_int16_array, as_int32_array, as_int64_array, as_int8_array, as_uint64_array,
+    };
+
+    fn with_fail_on_error<F: Fn(bool) -> Result<()>>(test_fn: F) {
+        for fail_on_error in [true, false] {
+            let _ = test_fn(fail_on_error);
```

Comment:
This would ignore the returned Result.

```suggestion
            let _ = test_fn(fail_on_error)?;
```

Reply:
Fixed. I couldn't add `?` but added `.expect`